src/ray/object_manager/plasma/store.cc

```diff
@@ -241,6 +241,18 @@ uint8_t *PlasmaStore::AllocateMemory(size_t size, MEMFD_TYPE *fd, int64_t *map_s
     }
   }
 
+  // Fallback to allocating from the filesystem.
+  if (pointer == nullptr && RayConfig::instance().plasma_unlimited()) {
```

Comment:
Even with threshold for object spilling (let's say 70%), we can unnecessarily create file based objects if spilling is not done quick enough right? Why don't we just move this code to when OOM actually happens? (that says if the space is not ready by object spilling by N seconds, we create file based objects)? I think we should be careful about creating too many of file-based objects.